tests/test_dle.py

```diff
@@ -523,6 +521,62 @@ def test_scheduling(self):
         assert not iterations[3].is_Affine
         assert 'schedule(dynamic,chunk_size)' in iterations[3].pragmas[0].value
 
+    @pytest.mark.parametrize('so', [0, 1, 2])
+    @pytest.mark.parametrize('dim', [1, 2])
```

Comment:
Is the `dim=0` case fix with the other part of that PR (the reduction isn't picked up if most outer loop).

Reply:
I'm gonna check now but I think so since the assert is now for all test cases

Reply:
yes, it looks correct:

```
    #pragma omp parallel num_threads(nthreads)
    {
      #pragma omp for collapse(3) schedule(static,1) reduction(+:f[0:f_vec->size[0]])
      for (int x = x_m; x <= x_M; x += 1)
      {
        for (int y = y_m; y <= y_M; y += 1)
        {
          for (int z = z_m; z <= z_M; z += 1)
          {
            f[y] += u[t0][x + 1][y + 1][z + 1] + 1;
          }
        }
      }
    }
```

Reply:
That's still the `y` dimension not `x`, need to change  `@pytest.mark.parametrize('dim', [0, 1, 2])` to get that previously problematic case.

Reply:
Added `0` back to dims and seems to be working fine now.